### ISSUE TITLE
Resizing view resizes the axis, but not the marks

### ISSUE THREAD

**Issue by sadruddin** [vega/vega#316]
When resizing a view, the axes get resized appropriately, but not the insides of the graph (marks). I tested the code below using the "bar" example spec

```javascript
vg.parse.spec(spec, function(chart) {
  view = chart({el:'#vis'});
  view.width(640).height(480).renderer('svg').update();
  view.width(1024).height(768).update({duration: 2000});
})
````

**Comment by arvind** [comment]
Yes, with the "bar" example, the rectangle marks are positioned and sized only on "enter" -- i.e. a data tuple for which a matching scene graph element does not exist. If the properties were instead moved to the "update" property set, the two commands above correctly resize the axes and bars for me. For more on the difference between enter and update, see [Thinking with Joins](http://bost.ocks.org/mike/join/) (although written for the D3 library, many of the concepts apply for Vega as well).

```json
    {
      "type": "rect",
      "from": {"data": "table"},
      "properties": {
        "enter": {
        },
        "update": {
          "x": {"scale": "x", "field": "x"},
          "width": {"scale": "x", "band": true, "offset": -1},
          "y": {"scale": "y", "field": "y"},
          "y2": {"scale": "y", "value": 0},
          "fill": {"value": "steelblue"}
        },
        "hover": {
          "fill": {"value": "red"}
        }
      }
    }
```

**Comment by sadruddin** [comment]
Indeed, thanks! It's a change of behaviour compared to 1.x, isn't it?

**Comment by arvind** [comment]
It shouldn't be. The Vega 1.5 [bar example](http://trifacta.github.io/vega/editor/?spec=bar) exhibits the same behaviour.

**Comment by sadruddin** [comment]
Indeed, now I realize I had removed the `props: 'enter'` parameter in my second update call, when I upgraded my code to 2.0 . However, when executing the code below on the respective demo editors, for the bar example, 1.5 and 2.0 do exhibit a different behaviour, the axis will disappear on 2.0, but will nicely scale on 1.5 . Not a problem for me, can work around that, but maybe it wasn't an intended change.

```javascript
ved.view.width(800).height(500).update({props : 'enter', duration: 1000})
```

**Comment by arvind** [comment]
That looks like a bug! Thanks for catching it.

**Comment by arvind** [comment]
Thanks for filing the issue, @sadruddin. This should be fixed with e86b067, and will be included in the next release of Vega (2.1, which is due early next week).

**Comment by sadruddin** [comment]
Excellent, looking forward to it!
